C/c4CAPI.cc

```diff
@@ -526,12 +526,15 @@ bool c4db_maintenance(C4Database* database, C4MaintenanceType type, C4Error *out
 
 // semi-deprecated
 C4Timestamp c4db_nextDocExpiration(C4Database *db) noexcept {
+    C4Error err;
+    returnIfCollectionInvalid(db->getDefaultCollection(), &err, C4Timestamp::Error);
```

Comment:
This doesn't solve the problem as calling db->getDefaultCollection() on the next line could still fail. You could get the collection, test it with `returnIfCollectionInvalid()` then use it on the next line.

```
auto collection = db->getDefaultCollection();
returnIfCollectionInvalid(collection, ...);
...
```